presto-base-jdbc/src/main/java/io/prestosql/plugin/jdbc/StandardColumnMappings.java

```diff
@@ -468,9 +467,10 @@ public static LocalTime fromPrestoTime(long value)
 
             case Types.CHAR:
             case Types.NCHAR:
-                // TODO this is wrong, we're going to construct malformed Slice representation if source > charLength
-                int charLength = min(columnSize, CharType.MAX_LENGTH);
-                return Optional.of(charColumnMapping(createCharType(charLength)));
+                if (columnSize > CharType.MAX_LENGTH) {
+                    return Optional.of(varcharColumnMapping(createUnboundedVarcharType()));
```

Comment:
why unbounded? Would bounded varchar, which has higher limit, be more appropriate here?

Reply:
Agree. Will add a TODO, as this should be test-covered

Reply:
Implemented as requested